### DISCUSSION TITLE
Keywords improvements

### DISCUSSION THREAD

**Discussion by SupinePandora43** [dotnet/csharplang#4434] [Language Ideas]
# Keywords in expression statements

```cs
void Test(Action action)
{
    action();
}
static void Main(){
    Test(() =>
        // invalid expression term 'unsafe'
        // ) expected
        // ; expected
        unsafe {
            Console.WriteLine("unsafe");
        }
    );
}
```

# Multiple keywords usage
```cs
static void Main(){
   unsafe fixed (byte* p = new byte[] { 1, 2, 3 })
   {
        Console.WriteLine("fixed pointer");
   }
}
```

# for loop example
```cs
static void Main(){
   unsafe fixed (byte* p = new byte[] { 1, 2, 3 }) for (int i = 0; i<3; i++)
   {
        Console.WriteLine(p[i]);
   }
}
```

**Comment by huoyaoyuan** [comment]
These scenarios are totally different restrictions.

> Keywords in expression statements

This is in fact **block for expression-bodied lambda**.

> Multiple keywords usage

This is **allowing unsafe block omitting braces**.

> for loop example

This too. `fixed` and `for` can be embedded in this way now.

**Reply by SupinePandora43** [reply]
```cs
   unsafe 
   {
       fixed (byte* p = new byte[] { 1, 2, 3 })
       {
            for (int i = 0; i<3; i++) {
                 Console.WriteLine(p[i]);
            }
       }
   }
```
?

**Reply by huoyaoyuan** [reply]
```csharp
using System;
public class C {
    static unsafe void Main()
    {
        fixed (byte* p = new byte[] { 1, 2, 3 }) for (int i = 0; i<3; i++)
        {
            Console.WriteLine(p[i]);
        }
    }
}
```

This is already valid. I mean `fixed` already has this ability.

**Reply by SupinePandora43** [reply]
~~i have no words, you won~~

"Keywords in expression statements" idea came to me from that:
![image](https://user-images.githubusercontent.com/36124472/107795518-90572500-6d7a-11eb-917d-9f07d42e9988.png)

~~just one less bracket~~
